### PR TITLE
fuel synax check

### PR DESCRIPTION
**Pull request by Zipfer**

### CODE REVIEW

**Comment by mihgen** [comment]
retest this please

File: utils/jenkins/fuel_syntax_check.sh [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#TODO: Run these in parallel - we have 4 cores.
+#TODO: Control the environment (through the config dir?).
+#      We want to parse for all environments.
+#      Is this being done, contrary to puppet report?
+#TODO: Even with --ignoreimport, some may be pulling in others,
+#      meaning we're checking multiple times.
+all_files=`find -name "*.pp" -o -name "*.erb" -o -name "*.rb" -o -name "*.sh"`
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh]: Please add allowed symbols check. Should be done in every bash script with such globs in use.

File: utils/jenkins/fuel_syntax_check.sh [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+num_files=`echo $all_files | wc -w`
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:12]: u no array count?

File: utils/jenkins/fuel_syntax_check.sh [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+if test $num_files -eq "0" ; then
+  echo "WARNING: no .sh, .pp, .rb or .erb files found"
+  exit 0
+fi
+echo "Checking $num_files files for syntax errors."
+echo "Puppet version is: `puppet --version`"
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:20]: if [ ! -z "$(which puppet)" ]; then foo ; else bar; fi

File: utils/jenkins/fuel_syntax_check.sh [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+if [ "$1" = "-u" ];
+then
+  all_files=`find -iname "rakefile"`
+  num_files=`echo $all_files | wc -w`
+  if test $num_files -eq "0" ; then
+    echo "WARNING: no Rakefile files found"
+    exit 0
+  fi
+  echo "Will run $num_files RSpec tests."
+  echo "Rake version is: `rake --version`"
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:55]: again, do the check please

File: utils/jenkins/fuel_syntax_check.sh [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  *.sh )
+    bash -n $x
+    ;;
+  esac
+done
+
+if [ "$1" = "-u" ];
+then
+  all_files=`find -iname "rakefile"`
+  num_files=`echo $all_files | wc -w`
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:49]: array count again

File: utils/jenkins/fuel_syntax_check.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    ;;
+  *.sh )
+    bash -n $x
+    ;;
+  esac
+done
+
+if [ "$1" = "-u" ];
+then
+  all_files=`find -iname "rakefile"`
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:48]: ARRAY=( $(expand_command) ), please fix this and one above

File: utils/jenkins/fuel_syntax_check.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    echo "WARNING: no Rakefile files found"
+    exit 0
+  fi
+  echo "Will run $num_files RSpec tests."
+  echo "Rake version is: `rake --version`"
+
+  for file in $all_files; do
+    d=`dirname $file`
+    dn=`basename $d`
+    cd $d
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:60]: quotes or check for spaces

File: utils/jenkins/fuel_syntax_check.sh [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+    rc=0
+    echo "RSpec-ing $dn"
+    set +e
+    rake spec
+    rc=$?
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:65]: expand error code at once in if

File: utils/jenkins/fuel_syntax_check.sh [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        --no-hard_tabs-check \
+        --with-filename $x
+
+    puppet parser validate --ignoreimport --render-as s --color=false $x
+    ;;
+  *.erb | *.rb )
+    erb -P -x -T '-' $x | ruby -c > /dev/null
+    ;;
+  *.sh )
+    bash -n $x
```
Comment (xdel) [inline utils/jenkins/fuel_syntax_check.sh:41]: why we`re checking exit codes below and not checking here?

**Comment by mihgen** [comment]
+1

**Pull request merged**